### PR TITLE
fix(layer): move backoff to spawned task

### PR DESCRIPTION
**Pull request by koivunej**
Move the backoff to spawned task as it can still be useful; make the sleep cancellable.

### CODE REVIEW

**Comment by koivunej** [comment]
Now waiting for the logging in #5766 to be merged first -- Still unclear if this is needed or not.

**Comment by koivunej** [comment]
Backoff has been useful with a recent prod issue. It should instead be moved to the spawned download task.

**Comment by koivunej** [comment]
> Backoff has been useful with a recent prod issue. It should instead be moved to the spawned download task.

Now implemented and reset to only [a0aa157](https://github.com/neondatabase/neon/pull/5746/commits/a0aa157e55f46556b6d7ea95d2f5e918125ff1a4).

**Comment by arpad-m** [comment]
Before merging this, a better commit message would be useful. This is not about all backoffs for example, but only a specific one. Also, some explanation about the why.

**Comment by koivunej** [comment]
> Before merging this, a better commit message would be useful. This is not about all backoffs for example, but only a specific one. Also, some explanation about the why.

The PR heading has "fix(layer): move backoff to spawned task". Layer only has one backoff. Where is the confusion?

File: pageserver/src/tenant/storage_layer/layer.rs
Comment (koivunej) [inline pageserver/src/tenant/storage_layer/layer.rs:876]: This was valid in #6134 but not in `main`, this list needs to have `timeline.cancel`.

**Review by arpad-m — APPROVED** [review]

File: pageserver/src/tenant/storage_layer/layer.rs
Comment (arpad-m) [inline pageserver/src/tenant/storage_layer/layer.rs:874]: Can this use tokio's Timeout functionality?

**Review by arpad-m — CHANGES_REQUESTED** [review]

File: pageserver/src/tenant/storage_layer/layer.rs
Comment (koivunej) [reply pageserver/src/tenant/storage_layer/layer.rs:874]: It could now, but I realized we need to cancel by the timeline cancellation token as well (similarly to #6133) in https://github.com/neondatabase/neon/pull/5746#discussion_r1428159294 since but of course #6134 has other TODO as well.

**Review by arpad-m — APPROVED** [review]

**Pull request merged**
